File: C++/034.cpp
#include <iostream>

using namespace std;

int sumFactorialDigits(int n) {
  int fac[] = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880};
  int sum = 0;
  while (n) {
    int mod = n%10;
    sum += fac[mod];
    n = (n-mod)/10;
  }
  return sum;
}

int main() {
  int sum = 0;
  for (int i=3; i<2540160; i++) {
    if (i == sumFactorialDigits(i))
      sum += i;
  }
  cout << sum;
}

File: Python/034.py
from math import factorial

def sum_factorial_digits(n):
    return sum([fac[int(i)] for i in str(n)])

fac = {i:factorial(i) for i in range(10)}
print(sum([i for i in range(3,2540160) if i == sum_factorial_digits(i)]))